### PR TITLE
chore(dbm): add dbm comment to database queries

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

This PR adds support for DBM comment injection in all ddtrace integrations.

 `dbapi.TracedCursor._trace_method` was updated to add DBM comments to sql queries executed in `[integration_name].query` spans. This adds DBM to the following integrations: `pyodbc`, `psycopg2`, `snowflake.connector`, `sqllite3`, `mariadb`, `mysql.connector`, `MySQLdb`, `pymysql`.

The following integrations do not use `dbapi.TracedCursor._trace_method` to trace database queries: `aiomysql`,`aiopg`, and `asyncpg`. They will be supported in a future PR

## Checklist
- [ ] Title must conform to [conventional commit](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional).
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [ ] Ensure tests are passing for affected code.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] PR cannot be broken up into smaller PRs.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [x] Backports are identified and tagged with Mergifyio.
- [x] Add to milestone.

### CODE REVIEW

File: tests/contrib/dbapi/test_dbapi.py [hunk 1]
```diff
@@ -30,6 +31,24 @@ def test_execute_wrapped_is_called_and_returned(self):
         assert "__result__" == traced_cursor.execute("__query__", "arg_1", kwarg1="kwarg1")
         cursor.execute.assert_called_once_with("__query__", "arg_1", kwarg1="kwarg1")
 
+    @TracerTestCase.run_in_subprocess(
+        env_overrides=dict(
+            DD_TRACE_SQL_COMMENT_INJECTION_MODE="full",
```
Comment (ZStriker19) [inline tests/contrib/dbapi/test_dbapi.py:36]: should we parametrize this and test the other `DD_TRACE_SQL_COMMENT_INJECTION_MODE` configs here?

**Review by majorgreys — COMMENTED** [review]
Should we make this change to just the dbapi integration in this PR?

File: ddtrace/settings/_database_monitoring.py [hunk 2]
```diff
@@ -34,9 +36,9 @@ class DatabaseMonitoringConfig(En):
 
 
 def _get_dbm_comment(db_span):
-    # type: (Span) -> str
+    # type: (Span) -> Optional[str]
     if dbm_config.injection_mode == "disabled":
-        return ""
+        return None
```
Comment (mabdinur) [inline ddtrace/settings/_database_monitoring.py:41]: Returning `None` makes more sense than an empty string. It's also more readable

File: tests/contrib/dbapi/test_dbapi.py (hunk 1 above)
Comment (mabdinur) [reply tests/contrib/dbapi/test_dbapi.py:36]: We will add integration tests to validate each injection mode

**Review by majorgreys — APPROVED** [review]

**Review by ZStriker19 — APPROVED** [review]

**Pull request merged**